preprocessors/openstreetmap/osm_service.py

```diff
@@ -314,79 +334,93 @@ def get_amenities(bbox_coord):
-                        if value != node.tags.get(
-                                "name") and value != node.tags.get("amenity"):
-                            if key not in amenity_record:
-                                amenity_record[key] = value
-
-                # Delete keys with no value
-                amenity_record = dict(
-                    x for x in amenity_record.items() if all(x))
-                amenity.append(amenity_record)
+                # Extract only amenities(under nodes) within the boundary
+                if node.lat >= lat_min and node.lat <= lat_max:
+                    if node.lon >= lon_min and node.lon <= lon_max:
+                        if node.tags.get("amenity") is not None:
+                            amenity_record = {
+                                "id": int(node.id),
+                                "lat": float(node.lat),
+                                "lon": float(node.lon),
+                                "name": node.tags.get("name"),
+                                "cat": node.tags.get("amenity"),
+                            }
+                            # Fetch as many tags possible beyond the basic
+
+                            for key, value in node.tags.items():
+                                if (value != node.tags.get(
+                                        "name") and
+                                        value != node.tags.get("amenity")):
+                                    if key not in amenity_record:
+                                        amenity_record[key] = value
+
+                        # Delete keys with no value
+                        amenity_record = dict(
+                            x for x in amenity_record.items() if all(x))
+                        amenity.append(amenity_record)
 
         if amenities.ways:
             for way in amenities.ways:
-                if way.tags.get("amenity") is not None:
-                    amenity_record = {
-                        "id": int(way.id),
-                        "lat": float(way.center_lat),
-                        "lon": float(way.center_lon),
-                        "name": way.tags.get("name"),
-                        "cat": way.tags.get("amenity"),
-                    }
-                    # Fetch as many tags possible
-                    for key, value in way.tags.items():
-                        if value != way.tags.get(
-                                "name") and value != way.tags.get("amenity"):
-                            if key not in amenity_record:
-                                amenity_record[key] = value
-                # Delete keys with no value
-                amenity_record = dict(
-                    x for x in amenity_record.items() if all(x))
-                amenity.append(amenity_record)
+                # Extract only amenities(under ways) within the boundary
+                if way.center_lat >= lat_min and way.center_lat <= lat_max:
```

Comment:
Could you include in your comment what `center_lat` and `center_lon` actually indicate here? If it's the center of an entire way, I'm not sure this is correct.

Reply:
Points of interest (e.g. amenities) can be found under three categories in OSM: nodes, ways and relations. So, a point of interest (e.g. a restaurant) may appear as a single point (i.e. as a node). However, there could also be a situation where different restaurants are combined to be one entity (i.e. a combination of nodes forming a way, or even a combination of ways for a relation). In that case, an overpass API uses "center_lat/center_lon" to determine the latitude and the longitude of the center of such a way or relation, https://python-overpy.readthedocs.io/en/latest/api.html#overpy.Relation.center_lat, So, in order to ensure that all the POIs in a given area are captured, one has to check those in the nodes, ways and relations, and get them extracted altogether. Kindly also note that the term "way" does not necessarily have to be street in OSM, building, structures, etc. can equally be categorized as ways (closed ways) in OSM, depending on the context of application.

https://wiki.openstreetmap.org/wiki/Elements#:~:text=Elements%20are%20of%20three%20types,how%20other%20elements%20work%20together)., https://wiki.openstreetmap.org/wiki/Node, https://wiki.openstreetmap.org/wiki/Way#:~:text=An%20OSM%20way%20is%20specified,figure%20of%20the%20Earth%20used, https://wiki.openstreetmap.org/wiki/Relation